SamlValve/src/main/java/com/axonivy/demo/saml/UserAuthenticator.java

```diff
@@ -0,0 +1,205 @@
+    if (!statusCode.equals("urn:oasis:names:tc:SAML:2.0:status:Success"))
+    {
+      throw new AuthenticationException("Invalid status code: " + statusCode);
+    }
+
+    if (response.getAssertions().size() != 1)
+    {
+      throw new AuthenticationException("The response doesn't contain exactly 1 assertion");
+    }
+
+    if (!response.getIssuer().getValue().equals("MySamlIssuer"))
+    {
+      throw new AuthenticationException("The response issuer didn't match the expected value");
+    }
+
+    // Add more Response validations here
+
+    return response.getAssertions().get(0);
+  }
+
+  private static void validateAssertion(Assertion assertion) throws AuthenticationException
+  {
+    if (!assertion.getIssuer().getValue().equals("MySamlIssuer"))
+    {
+      throw new AuthenticationException("The assertion issuer didn't match the expected value");
+    }
+
+    if (assertion.getSubject().getNameID() == null)
+    {
+      throw new AuthenticationException("The NameID value is missing from the SAML response");
+    }
+
+    // Add more Assertion validations here
+
+    enforceConditions(assertion.getConditions());
+  }
+
+  private static void validateSignature(Assertion assertion) throws AuthenticationException
+  {
+    try
+    {
+      Credential credential = new BasicX509Credential(SIGN_VERIFICATION_CERT);
+      SignatureValidator.validate(assertion.getSignature(), credential);
+    }
+    catch (SignatureException se)
+    {
+      throw new AuthenticationException("Error verifying assertion signature", se);
+    }
+  }
+
+  private static void enforceConditions(Conditions conditions) throws AuthenticationException
+  {
+    DateTime now = DateTime.now();
+
+    if (now.isBefore(conditions.getNotBefore()))
+    {
+      throw new AuthenticationException(
+              "Assertion cannot be used before " + conditions.getNotBefore().toString());
+    }
+
+    if (now.isAfter(conditions.getNotOnOrAfter()))
+    {
+      throw new AuthenticationException(
+              "Assertion cannot be used after  " + conditions.getNotOnOrAfter().toString());
+    }
+
+    // Enforce more Assertion Conditions here
+  }
+
+  private static X509Certificate loadCertificate(String filePath)
+  {
+    try (InputStream is = new FileInputStream(filePath))
+    {
+      CertificateFactory fact = CertificateFactory.getInstance("X.509");
+      return (X509Certificate) fact.generateCertificate(is);
+    }
+    catch (Exception e)
+    {
+      throw new RuntimeException(e);
+    }
+  }
+
+  protected static String decodeSamlResponse(Request request) throws AuthenticationException
+  {
+    Optional<String> base64SamlToken = extractBase64SamlToken(request);
+    return base64SamlToken
+            .map(token -> new String(Base64.getDecoder().decode(token), StandardCharsets.UTF_8))
+            .orElseThrow(() -> new AuthenticationException("No saml assertion provided for request " + request.getRequestURI()));

```

Comment:
because of this. the engine can only be accessed via saml. 
I think if its not a saml request, you should be still able to login without any SSO mechanism.